uiowa/uiowa
# Create uiowa_profiles module and split for integration with profiles client.

Resolves https://github.com/uiowa/uiowa/issues/3286


### How to test

1. Check out and pull down `i3286-directory-profiles`.
2. Sync down the `default` site.
3. Run `drush @default.local uli`.
4. Enable the Profiles split and import configuration.
5. Navigate to https://default.local.drupal.uiowa.edu/admin/config and, under the `System` header, click `UIowa Profiles Integration`.
6. create a new profiles instance using the `+` button
7. for each profiles instance:  
    7.1. use either of these API keys here for separate instances to check if multiple profiles pages work: https://iowaweb.slack.com/archives/C0164FNN1PV/p1623858986410400?thread_ts=1623858978.410300&cid=C0164FNN1PV.
    7.2. fill out the rest of the form and save configuration
    7.3. Go to the path you previously defined.
    7.4. You should see a Directory profiles listing for microbiology, as that is the API key we have used as a test.
8. verify that you can safely delete profiles instances
9. try and break things.

### Todo
- [x] Move to feature split
- [x] Decide whether or not to render the Drupal page title and breadcrumbs and adjust accordingly. (Passing in page title and breadcrumbs to Vue app may not be supported yet.)
- [x] Fix page size config (Passing this in to the Vue app is not supported yet.)
- [x] Render intro config on directory page (Passing intro config into Vue app is not supported yet.)
  - We can try to render this ourselves but then we'd need to hide it on person pages which could be difficult.
- [x] Fix 404 on individual profile page refresh
- [x] Fix sitemap
- [x] Misc. styling 
- [x] Decide what to do with the canonical link config.
- [x] Split webmaster permissions.
